c++/src/Reader.cc

```diff
@@ -854,6 +854,8 @@ namespace orc {
 
   class ReaderImpl : public Reader {
   private:
+    const int64_t epochOffset;
+
```

Comment:
The ColumnReader uses StripeStreamsImpl::getEpochOffset() and the implementation of that method uses ReaderImpl::epochOffset. The initializer for ReaderImpl::epochOffset uses the static method getEpochOffset, but it should only be called once per a reader.